test/gurka/test_landmarks.cc

```diff
@@ -327,55 +334,61 @@ TEST(LandmarkTest, TestTileStoreLandmarks) {
   GraphReader gr(landmark_map.config.get_child("mjolnir"));
   auto tile = gr.GetGraphTile(tile_id);
 
+  auto check_landmark = [&](const auto& landmark, const auto& point) {
+    auto ltype = static_cast<LandmarkType>((edge_id.id() + 1) % invalid_landmark);
+
+    const double rounding_error = 5e-7;
+    // we dont store this in the tile
+    EXPECT_EQ(landmark.id, 0);
+    // opposing edges will have a copy of the edge we associated it to so we need to check it
+    bool skip = false;
+    if (landmark.name != std::to_string(edge_id)) {
+      auto opp = gr.GetOpposingEdgeId(edge_id);
+      ASSERT_EQ(landmark.name, std::to_string(opp)) << "Unexpected landmark name or edge/opp_edge";
+      skip = true;
+    }
+    // we dont do deep check on opposing edge records
```

Comment:
previously this test was relying on luck to pass. in truth we will find the landmark both on its edge and its opposing edge (since they share edgeinfo), when its the opposing copy we skip the more difficult tests saving them only for the direction of the edge where we named it to match the edge id